surrealdb/clients/ws.py

```diff
@@ -0,0 +1,284 @@
+
+    async def __aenter__(self) -> "SurrealDBWSClient":
+        await self.connect()
+        return self
+
+    async def __aexit__(
+        self,
+        exc_type: Optional[Type[BaseException]] = None,
+        exc_value: Optional[BaseException] = None,
+        traceback: Optional[TracebackType] = None,
+    ) -> None:
+        await self.disconnect()
+
+    async def connect(self) -> None:
+        self._client = ClientSession()
+        self._ws = await self._client.ws_connect(self._url)
+
+        self._recv_task = asyncio.create_task(self._receive_task())
+        self._recv_task.add_done_callback(self._receive_complete)
+
+        if self._token is not None:
+            await self.authenticate(self._token)
+
+        if self._username is not None and self._password is not None:
+            await self.sign_in(username=self._username, password=self._password)
+
+        if self._namespace is not None and self._database is not None:
+            await self.use(self._namespace, self._database)
+
+    async def disconnect(self) -> None:
+        self._recv_task.cancel()
+
+        await self._ws.close()
+        await self._client.close()
+
+    def _receive_complete(self, task: asyncio.Task) -> None:
+        try:
+            task.result()
+        except asyncio.CancelledError:
+            pass
+        except Exception as e:
+            raise e
+
+    async def _receive_task(self) -> None:
+        async for msg in self._ws:
+            if msg.type == WSMsgType.ERROR:
+                raise SurrealWebsocketException(msg.data)
+
+            response: RPCResponse = jsonlib.loads(msg.data)
+            if response.get("error") is not None:
+                raise SurrealWebsocketException(response["error"]["message"])
+
+            self._responses[response["id"]] = response
+
+    async def _wait_response(self, id: str) -> RPCResponse:
+        while id not in self._responses:
+            await asyncio.sleep(0.1)
+
+        response = self._responses.pop(id)
+        return response
+
+    async def _send(
+        self,
+        method: str,
+        *params: Any,
+    ) -> Any:
+        request: RPCRequest = {
+            "id": generate_id(length=16),
+            "method": method,
+            "params": params,
+        }
+
+        await self._ws.send_json(request)
+
+        response = await self._wait_response(request["id"])
+        return response["result"]
+
+    async def ping(self) -> bool:
+        response = await self._send("ping")
+        return response
+
+    async def use(
+        self,
+        namespace: Optional[str] = None,
+        database: Optional[str] = None,
+    ) -> None:
+        response = await self._send("use", namespace, database)
+
+        self._namespace = namespace
+        self._database = database
+
+        return response
+
+    async def info(self) -> None:
+        response = await self._send("info")
+        return response
+
+    async def sign_up(
```

Comment:
```suggestion
    async def signup(
```
No need for an `_` really